### ISSUE TITLE
Bug: prototype collusion in new-cap rule

### ISSUE THREAD

**Issue by swwind** [eslint/eslint#21255]
### Environment

Node version: v26.3.0
npm version: v11.16.0
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: linux 7.0.0-29-generic

### What parser are you using?

Default (Espree)

### What did you do?

```js
/*eslint new-cap: "error"*/

// ok
new lowerCase()   // reports
new UpperCase()  // not reports

// false-negative
new toString()  // not reports
new constructor()  // not reports
new hasOwnProperty()  // and so ...
new isPrototypeOf()
new propertyIsEnumerable()
new valueOf()
new toLocaleString()
```

### What did you expect to happen?

`new toString()` and `new constructor()` alike should be reported.

### What actually happened?

omg, it did not report.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbmV3LWNhcDogXCJlcnJvclwiKi9cblxuLy8gb2tcbm5ldyBsb3dlckNhc2UoKSAgIC8vIHJlcG9ydHNcbm5ldyBVcHBlckNhc2UoKSAgLy8gbm90IHJlcG9ydHNcblxuLy8gZmFsc2UtbmVnYXRpdmVcbm5ldyB0b1N0cmluZygpICAvLyBub3QgcmVwb3J0c1xubmV3IGNvbnN0cnVjdG9yKCkgIC8vIG5vdCByZXBvcnRzXG5uZXcgaGFzT3duUHJvcGVydHkoKSAgLy8gYW5kIHNvIC4uLlxubmV3IGlzUHJvdG90eXBlT2YoKVxubmV3IHByb3BlcnR5SXNFbnVtZXJhYmxlKClcbm5ldyB2YWx1ZU9mKClcbm5ldyB0b0xvY2FsZVN0cmluZygpIiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319LCJlY21hVmVyc2lvbiI6ImxhdGVzdCJ9fX0=

### Participation

- [ ] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Apparently there is some raw property checks to an object. Prefer `Object.create(null)` or `Object.hasOwn(map, "key")`.

**Comment by crimsonjay0** [comment]
I'd like to be assigned to this.

**Comment by mdjermanovic** [comment]
Thanks for the report. I agree this is a bug, PR is welcome. Assigning @crimsonjay0 as requested.
